assignment/week2/가계부/index.js

```diff
@@ -0,0 +1,214 @@
+const INIT_BALANCE = 0;
+const HISTORY_LIST = [
+  {
+    id: 1,
+    history: '식비',
+    place: '맥도날드 숭실대점',
+    money: 'spend',
+    amount: '6900',
+  },
+  {
+    id: 2,
+    history: '월급',
+    place: '평촌 김장현 수학',
+    money: 'income',
+    amount: '500000',
+  },
+  {
+    id: 3,
+    history: '구독',
+    place: '쿠팡',
+    money: 'spend',
+    amount: '4990',
+  },
+  {
+    id: 4,
+    history: '쇼핑',
+    place: '지그재그',
+    money: 'spend',
+    amount: '27000',
+  },
+];
+
+document.addEventListener('DOMContentLoaded', () => {
+  displayList(HISTORY_LIST);
+  displayMyAsset(HISTORY_LIST);
+});
+
+function displayList(item) {
+  item.map((item) => createList(item));
+}
+
+function createList(item) {
+  const ul = document.getElementById('contents_list');
+  const li = document.createElement('li');
+  li.className = 'contents';
+  li.id = item.id;
+  const content1 = document.createElement('div');
+  const content2 = document.createElement('div');
+  const history = document.createElement('p');
+  history.className = 'history';
+  const place = document.createElement('p');
+  place.className = 'place';
+  const amount = document.createElement('p');
+  const closeBtn = document.createElement('i');
+  closeBtn.className = 'closeBtn';
+
+  closeBtn.addEventListener('click', (event) => checkAndDeleteList(event));
+
+  history.textContent = item.history;
+  place.textContent = item.place;
+  closeBtn.innerHTML = '<i class="fa-solid fa-xmark"></i>';
+
+  if (item.money == 'spend') {
+    amount.className = 'spending';
+    amount.textContent = `-${comma(item.amount)}`;
+  } else {
+    amount.className = 'income';
+    amount.textContent = `+${comma(item.amount)}`;
+  }
+
+  ul.appendChild(li);
+  li.appendChild(content1);
+  li.appendChild(content2);
+  content1.appendChild(history);
+  content1.appendChild(place);
+  content2.appendChild(amount);
+  content2.appendChild(closeBtn);
+}
+
+function displayMyAsset(item) {
+  const asset_box = document.getElementById('my_money');
+  const income_box = document.getElementById('in');
+  const spend_box = document.getElementById('out');
+  let out_ = 0;
+  let in_ = 0;
+
+  item.map((item) => {
+    if (item.money == 'spend') {
+      out_ += Number(uncomma(item.amount));
+    } else if (item.money == 'income') {
+      in_ += Number(uncomma(item.amount));
+    }
+  });
```

Comment:
```suggestion
  item.map(({money, amount}) => {
    if (money === 'spend') {
      out_ += Number(uncomma(amount));
    } else if (money === 'income') {
      in_ += Number(uncomma(amount));
    }
  });
```
구조분해할당을 하니 훨씬 가독성이 좋아지죠! switch문 사용하는 것도 좋을 수 있을 것 같아요 ㅎㅎ